packages/components/src/components/back-to-top/back-to-top.tsx

```diff
@@ -0,0 +1,55 @@
+import { Component, h, State, Prop, Listen } from '@stencil/core';
+import { CssClassMap } from '../../utils/utils';
+import classNames from 'classnames';
+
+@Component({
+  tag: 't-back-to-top',
+  styleUrl: 'back-to-top.css',
+  shadow: true,
+})
+export class BackToTop {
+  @Prop() public visibilityHeight?: number = 200;
+
+  @State() private visible: boolean = false;
+
+  @Listen('scroll', { target: 'window' })
+  public handleScroll(ev) {
+    if (
+      (ev.target.documentElement &&
+        ev.target.documentElement.scrollTop > this.visibilityHeight) ||
+      (ev.target.body && ev.target.body.scrollTop > this.visibilityHeight)
+    ) {
+      this.visible = true;
+    } else {
+      this.visible = false;
+    }
+  }
+
+  public scrollToTop = () => {
+    window.scrollTo({
+      top: 0,
+      left: 0,
+      behavior: 'smooth',
+    });
+  };
+
+  public render() {
+    this.visible = document.documentElement.scrollTop > this.visibilityHeight;
+    return (
+      <div
+        class={this.getCssClassMap()}
+        onClick={this.scrollToTop}
+        onScroll={() => this.handleScroll}
+      >
+        <slot />
```

Comment:
The slot is not very usable right now as the `css` is not compatible. If there is a slot, the arrow is still displayed and the scroll button is not adapting the slot content. 
![image](https://user-images.githubusercontent.com/27810942/72056070-2db86d80-32cc-11ea-8445-d786c82de8a0.png)